platform/linux-generic/odp_random_std.c

```diff
@@ -58,21 +58,21 @@ static int32_t _random_data(uint8_t *buf, uint32_t len, uint64_t *seed)
 	}
 
 	for (uint32_t i = 0; i < len / 4; i++) {
-		*(uint32_t *)(uintptr_t)buf = xorshift64s32(seed);
+		*(odp_una_u32_t *)buf = xorshift64s32(seed);
```

Comment:
Not assuming alignment here can throw away the work done to align the pointer if _ODP_UNALIGNED is zero. If the CPU does not support unaligned accesses, the compiler may have to resort to using multiple store instructions here, even though it is not really necessary.

I would suggest doing something like this here:

```
for (uint32_t i = 0; i < len / 4; i++) {
        if (_ODP_UNALIGNED)
                *(uint32_t *)(uintptr_t)buf = xorshift64s32(seed);
        else
                *(odp_una_u32_t *)buf = xorshift64s32(seed);
        buf += 4;
}
```

Reply:
I did it a bit differently, in v4. But maybe there's no real need to optimize the no-unaligned case, and we could just remove the whole thing?

Reply:
Maybe, but since it is already there and you fixed the alignment issues, I think it can as well stay.